docs/proposals/20210722-openyurt-convertor-operator.md

```diff
@@ -0,0 +1,175 @@
+
+## Table of Contents
+* [Convert Vanilla Kubernetes cluster to OpenYurt cluster through Opertaor](#convert-vanilla-kubernetes-cluster-to-openyurt-cluster-through-operator)
+   * [Table of Contents](#table-of-contents)
+   * [Glossary](#glossary)
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals/Future Work](#non-goalsfuture-work)
+   * [Proposal](#proposal)
+      * [User Stories](#user-stories)
+         * [Story 1](#story-1)
+         * [Story 2](#story-2)
+      * [Requirements](#requirements)
+         * [Functional Requirements](#functional-requirements)
+            * [FR1](#fr1)
+            * [FR2](#fr2)
+            * [FR3](#fr3)
+         * [Non-Functional Requirements](#non-functional-requirements)
+            * [NFR1](#nfr1)
+            * [NFR2](#nfr2)
+      * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Risks and Mitigations](#risks-and-mitigations)
+   * [Alternatives](#alternatives)
+   * [Upgrade Strategy](#upgrade-strategy)
+   * [Additional Details](#additional-details)
+      * [Test Plan](#test-plan)
+   * [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+
+This OpenYurt Convertor Operator is to translate a vanilla Kubernetes cluster into an OpenYurt cluster, through a simple API (YurtConvetor CRD).
+
+## Motivation
+
+To make OpenYurt translation easy for user, i.e., as automative as possible.
+
+### Goals
+- Convert nodes automatically
+- Revert nodes automatically
+- Translated OpenYurt cluster should be the same as through Yurtcutl or doing manually
+
+### Non-Goals/Future Work
+
+This is NOT a replacement of yurtctl command.
+
+## Proposal
+
+This is an operator, with CRD and Controller.
+
+CRD's definition:
+
+// YurtconvertorSpec defines the desired state of Yurtconvertor
+type YurtconvertorSpec struct {
+        CloudNodes               []string `json:"cloudnodes,omitempty"`
+        Convert                  []string `json:"convert,omitempty"`
+        Revert                   []string `json:"revert,omitempty"`
+        Except                   []string `json:"except,omitempty"`
+        EnableYurtTunnel         bool     `json:"enable-yurt-tunnel,omitempty"`
+        EnableYurtAppManager     bool     `json:"enable-yurt-app-manager,omitempty"`
+        YurtTunnelVersion        string   `json:"yurt-tunnel-version,omitempty"`
+        YurtAppManagerVersion    string   `json:"yurt-app-manager-version,omitempty"`
+}
+
+// YurtconvertorStatus defines the observed state of Yurtconvertor
+type YurtconvertorStatus struct {
+        Converted []string `json:"converted,omitempty"`
+        Reverted  []string `json:"reverted,omitempty"`
+        Excepted  []string `json:"excepted,omitempty"`
+        Failed    []string `json:"failed,omitempty"`
+}
```

Comment:
Please update the Status based on the SPEC modifications suggested above accordingly. 
For failed nodes, the names are not enough. We need to expose the error msgs in the Status as well.